iac-common/src/testFixtures/java/org/sonar/iac/common/testing/AbstractExternalRulesDefinitionTest.java

```diff
@@ -0,0 +1,106 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.common.testing;
+
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.sonar.api.SonarProduct;
+import org.sonar.api.SonarRuntime;
+import org.sonar.api.server.rule.RulesDefinition;
+import org.sonar.iac.common.reports.AbstractExternalRulesDefinition;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+import static org.sonar.iac.common.testing.IacTestUtils.SONARLINT_RUNTIME_9_9;
+import static org.sonar.iac.common.testing.IacTestUtils.SONAR_RUNTIME_10_6;
+import static org.sonar.iac.common.testing.IacTestUtils.SONAR_RUNTIME_9_9;
+
+public abstract class AbstractExternalRulesDefinitionTest {
+
+  static Stream<Arguments> externalRepositoryShouldBeInitializedWithSonarRuntime() {
+    return Stream.of(
+      Arguments.of(SONAR_RUNTIME_10_6, true),
+      Arguments.of(SONAR_RUNTIME_9_9, false),
+      Arguments.of(SONARLINT_RUNTIME_9_9, false));
+  }
+
+  @MethodSource
+  @ParameterizedTest
+  void externalRepositoryShouldBeInitializedWithSonarRuntime(SonarRuntime sonarRuntime, boolean shouldSupportCCT) {
+    var context = new RulesDefinition.Context();
+    AbstractExternalRulesDefinition rulesDefinition = rulesDefinition(sonarRuntime);
+
+    if (sonarRuntime.getProduct() != SonarProduct.SONARLINT) {
+      rulesDefinition.define(context);
+      assertExistingRepository(context, rulesDefinition, shouldSupportCCT);
+    } else {
+      assertNoRepositoryIsDefined(context, rulesDefinition);
+    }
+
+  }
+
+  protected void assertExistingRepository(
+    RulesDefinition.Context context,
+    AbstractExternalRulesDefinition rulesDefinition,
+    boolean shouldSupportCCT) {
+    assertThat(context.repositories()).hasSize(1);
+    var repository = context.repository("external_" + reportKey());
+    assertThat(repository).isNotNull();
+    assertThat(repository.name()).isEqualTo(reportName());
+    assertThat(repository.language()).isEqualTo(language());
+    assertThat(repository.isExternal()).isTrue();
+    assertThat(repository.rules()).hasSize(numberOfRules());
+
+    for (String ruleKey : rulesDefinition.getRuleLoader().ruleKeys()) {
+      assertThatNoException().isThrownBy(() -> rulesDefinition.getRuleLoader().ruleSeverity(ruleKey));
+      assertThatNoException().isThrownBy(() -> rulesDefinition.getRuleLoader().ruleType(ruleKey));
+    }
+
+    assertThat(rulesDefinition.getRuleLoader().isCleanCodeImpactsAndAttributesSupported()).isEqualTo(shouldSupportCCT);
+  }
+
+  @Test
+  public void noOpRulesDefinitionShouldNotDefineAnyRule() {
```

Comment:
As written in another comment this method is not executed when calling `./gradlew :iac-common:test`.